modules/ROOT/pages/cryptography-jce.adoc

```diff
@@ -7,53 +7,99 @@ The JCE strategy enables you to use the wider range of cryptography capabilities
 
-.Example: PBE Encryption
+=== Configure Password-Based Encryption From Anypoint Studio
+
+To configure PBE from Anypoint Studio, follow these steps:
+
+. From the Mule palette, add the Crypto module to your project.
++
+See xref:cryptography.adoc#install-crypto-module[Install the Extension] for instructions.
+. Select the desired operation, and drag the component to the flow:
++
+image::mruntime-crypto-pbe-add.png[crypto-pbe-add]
+. In the component view, configure the *Algorithm* and *Password* properties:
++
+image::mruntime-crypto-pbe-config.png[crypto-pbe-config]
+
+=== XML Examples
+
+The following are XML examples for each each of the PBE operations:
+
+* Example: PBE Encryption
++
 [source,xml,linenums]
 ----
 <crypto:jce-encrypt-pbe password="a-Sup3r_Secure-Passw0rd"/>
 ----
++
 If no algorithm is specified, `PBEWithHmacSHA256AndAES_128` is used.
 
-.Example: PBE Decryption
+* Example: PBE Decryption
++
 [source,xml,linenums]
 ----
 <crypto:jce-decrypt-pbe algorithm="PBEWithHmacSHA256AndAES_128" password="a-Sup3r_Secure-Passw0rd"/>
 ----
 
-.Example: PBE Signature
+* Example: PBE Signature
++
 [source,xml,linenums]
 ----
 <crypto:jce-sign-pbe password="a-Sup3r_Secure-Passw0rd"/>
 ----
-
++
 If no algorithm is specified, `PBEWithHmacSHA256` is used.
 
-.Example: PBE Signature Validation
+* Example: PBE Signature Validation
++
 [source,xml,linenums]
 ----
 <crypto:jce-validate-pbe password="a-Sup3r_Secure-Passw0rd" algorithm="PBEWithHmacSHA256" expected="#[vars.expectedSignature]"/>
 ----
-
++
 The `expected` parameter defines the signature used to validate the message.
 
 == Key-Based Encryption
 
-This section provides key-based encryption examples.
+Configure a symmetric or asymmetric key to perform encryption/signing operations.
+
+=== Configure Key-Based Encryption From Anypoint Studio
 
-=== Configuration
+To configure Key-based encryption operations from Anypoint Studio, follow these steps:
+
+. From the Mule palette, add the Crypto module to your project.
++
+See xref:cryptography.adoc#install-crypto-module[Install the Extension] for instructions.
+. Select the desired operation, and drag the component to the flow:
++
+image::mruntime-crypto-jce-add.png[crypto-jce-add]
+. Open the component properties and select an existing *Module configuration*, or create a new one by specifying the *Keystore*, *Type* (JKS, JCEKS, PKCS12) and *Password*.
++
+You can also add Symmetric or Asymmetric key info to be used in the sign operations:
++
+image::mruntime-crypto-jce-global-config.png[crypto-jce-global-config]
+. Configure *Key selection* by using a *Key id* previously defined in the Module configuration, or define a new one for this operation:
++
+image::mruntime-crypto-jce-config.png[crypto-jce-config]
+. Select the Algorithm to use during the operation.
+
+=== XML Examples
+
+The following XML examples show a JCE configuration that defines symmetric and asymmetric keys and different operations using these keys.
+
+==== Configuration
```

Comment:
In general, we want to avoid nesting text down four or more levels.
When this happens, reevaluate topic organizations and see if it is possible to uplevel information.